pkg/util/webhookpatch.go

```diff
@@ -34,33 +30,17 @@ func PatchMutatingWebhookConfig(client admissionregistrationv1beta1client.Mutati
 	if err != nil {
 		return err
 	}
-	prev, err := json.Marshal(config)
-	if err != nil {
-		return err
-	}
 	found := false
 	for i, w := range config.Webhooks {
 		if w.Name == webhookName {
 			config.Webhooks[i].ClientConfig.CABundle = caBundle
 			found = true
-			break
 		}
 	}
 	if !found {
 		return apierrors.NewInternalError(fmt.Errorf(
 			"webhook entry %q not found in config %q", webhookName, webhookConfigName))
 	}
-	curr, err := json.Marshal(config)
-	if err != nil {
-		return err
-	}
-	patch, err := strategicpatch.CreateTwoWayMergePatch(prev, curr, v1beta1.MutatingWebhookConfiguration{})
-	if err != nil {
-		return err
-	}
-
-	if string(patch) != "{}" {
-		_, err = client.Patch(context.TODO(), webhookConfigName, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
-	}
+	_, err = client.Update(context.TODO(), config, metav1.UpdateOptions{})
```

Comment:
Is it equivalent with the old code ? Better ?

Reply:
Its much simpler than the old code and consistent with what we are doing in other parts of the code (including validation webhook). I couldn't get the patch to work with multiple webhooks - not sure where the bug lies exactly